### DISCUSSION TITLE
27. September 2022

### DISCUSSION THREAD

**Discussion by dimakuv** [gramineproject/gramine#925] [Meeting notes]
### Agenda

*(please write your proposed agenda items in comments under this discussion)*

- [x] Benny Fuhry: Discussion about documentation restructuring

### Misc

Dmitrii, Michal and Woju had to finish one PR in the `contrib` repo, which took 15 minutes of the meeting.

### Benny Fuhry: Discussion about documentation restructuring

- Gramine documentation is in ReadTheDocs as an offering (not on our server); is it what we want?
    - Woju: this is popular, especially with Python developers. Woju is not aware of anyone running ReadTheDocs on the server.
	- Benny: who has the admin rights to the page?
	    - Woju and Dmitrii (the technical writer will contact them).
    - The technical writer wants to change the configurations -- version of Sphinx, some extensions.
	- Woju: we pin to a specific version of Sphinx, because on some older systems, newer versions of Sphinx had bugs. Unclear if this is still the case, and probably Sphinx may be updated now.
	
- In the long run, the technical writer will be open to reviewing changes in the documentation.
    - Woju and Michal: this will be very good.
	
- Website on WordPress:
    - Michal requests a set of plugins that will be required. Many plugins are known to be vulnerable, and they are not fixed.
	- Michal and Mariusz suggest to have a static HTML render of the WordPress contents (there are tools for this). The static HTML will be hosted on the public server and won't be vulnerable. The dynamic WordPress will run in some sandboxed environment (e.g. behind the corporate proxy), where technical writers can use their flows. This also has clear separation: only Gramine maintainers have access to the public server, and technical writers have access only to the sandboxed environment.
	- Don: need to talk to the technical writer. Whether (s)he is fine with the sandboxed WordPress instance where (s)he can do the work, and then give the results to Gramine maintainers that will run it through the static tool and upload on the public website.
	- Benny: will talk to the technical writer and maybe bring her/him to the core Gramine meeting.

**Comment by BFuhry** [comment]
Discussion about documentation restructuring

**Reply by woju** [reply]
@BFuhry as a followup, I have the notes about what we have attached.
[docs-inventory.txt](https://github.com/gramineproject/gramine/files/9665682/docs-inventory.txt)
